servicebroker/test/model/serviceinstance_test.js

```diff
@@ -1,35 +1,71 @@
 'use strict';
 var expect = require('chai').expect;
+var serviceInstance = require('../../lib/models')().service_instance;
+
+var fake_serviceId = "test_service";
+var fake_orgId = "test_org";
+var fake_spaceId = "test_space";
+var fake_orgId2 = "test_org2";
+var fake_spaceId2 = "test_space2";
+var fake_condition = {
+  'serviceId': fake_serviceId,
+  'orgId': fake_orgId,
+  'spaceId': fake_spaceId,
+  where: { 'serviceId': fake_serviceId, 'orgId': fake_orgId, 'spaceId': fake_spaceId }
+};
+var fake_condition2 = {
+  'serviceId': fake_serviceId,
+  'orgId': fake_orgId2,
+  'spaceId': fake_spaceId2,
+  where: { 'serviceId': fake_serviceId, 'orgId': fake_orgId2, 'spaceId': fake_spaceId2 }
+};
 
-var fake_serviceinstance = { 'serviceId':'test_service', 'orgId':'test_org', 'spaceId':'test_space' };
 
 describe('Service_instance Model Definition Test Suite', function() {
+
   before(function() {
-    this.serviceInstance = require('../../lib/models')().service_instance;
-    this.serviceInstance.sequelize.sync();
+    serviceInstance.sequelize.sync();
+    serviceInstance.truncate({ cascade: true });
   });
-  
+
   beforeEach(function() {
-    this.serviceInstance.truncate({ cascade: true });
+    serviceInstance.truncate({ cascade: true });
+  });
+
+  it('Should create a new service instance successfully with the 1st findOrCreate', function(done) {
+    serviceInstance.findOrCreate(fake_condition)
+      .then(function(result) {
+        expect(result[0].serviceId).to.equal(fake_condition.serviceId);
+        expect(result[1]).to.equal(true);
+        done();
+      });
   });
 
-  it('Should create a service instance successfully', function() {
-    return this.serviceInstance.create(fake_serviceinstance )
-      .then(function(instance) {
-        expect(instance.serviceId).to.equal(fake_serviceinstance.serviceId);
+
+  it('Should find the existing service instance successfully with the 2nd findOrCreate', function(done) {
+    serviceInstance.findOrCreate(fake_condition)
+      .then(function(result) {
+        serviceInstance.findOrCreate(fake_condition)
+          .then(function(result) {
+            expect(result[0].serviceId).to.equal(fake_condition.serviceId);
+            expect(result[1]).to.equal(false);
+            done();
+          });
+      });
+  });
+
+  it('Should throw UniqueConstraintError with conflict record', function(done) {
+    serviceInstance.findOrCreate(fake_condition)
+      .then(function(result) {
+        serviceInstance.findOrCreate(fake_condition2)
+          .then(function(result) {})
```

Comment:
we should return promise here instead of calling done. Mocha documents "Alternately, instead of using the done() callback, you may return a Promise. This is useful if the APIs you are testing return promises instead of taking callbacks".   
We are blocked here for about 1 hour by missing "done" or "return" here.  The cmd "mocha test/models/serviceinstance_test" passed, and the cmd " mocha test/routes/serviceinstance_test" passed as well, but the cmd "mocha test --recuisive" sometimes failed in "test/routes/serviceinstance_test"  with error "done() callback is not handled correctly".  It did take time to figure out the done is missing in models/serviceinstance_test ...  

I would like to follow the "return" approach to handle promise as suggested by the document